.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,45 @@
+name: "CodeQL"
+
+on:
+  push:
+  pull_request:
+    branches: [ main ]
+    paths-ignore:
+      - '**.md'
+      - 'docs/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java' ]
```

Comment:
Why a matrix for one element?

Reply:
This is the template generated by Github when enabling CodeQL. It is a matrix so that it is easy to add additional languages. Although not required in our case I think it does not hurt either, and I would leave it for consistency with EDC.